statsd/taggable.go

```diff
@@ -83,49 +81,6 @@ func WatchingTaggable(ctx context.Context, t Taggable) context.Context {
 	return &taggableContext{Context: ctx, taggable: t}
 }
 
-// WithTagLogFields combines logger.WithFields(fields) and WithTags(tags)
-// This simplifies the common operation of adding fields to the logger and the metrics
-// This argument purposefully not typed as Tags, such that logrus.Fields and Tags can both be passed without additional casting.
-func WithTagLogFields(ctx context.Context, tags map[string]interface{}) context.Context {
-	ctx = logger.WithFields(ctx, tags)
-	ctx = WithTags(ctx, tags)
-	return ctx
-}
-
-// WithTagLoggable combines WithTaggable and logger.WithLoggable
-// If the Loggable is a Taggable already (implements StatsTags), it will be used directly.
-// If StatsTags doesn't exist, LogFields() will be used instead.
-func WithTagLoggable(ctx context.Context, l logger.Loggable) context.Context {
```

Comment:
unless what you mean is that fields implements loggable and taggle?

Reply:
@hurracrane I meant that the API will first do `ctx = logger.WithLoggable(ctx, shop)` Then it will try to cast it as Taggable and add it if so. 

Basically avoid the fallback of directly adding LogFields as tags. 

But I'm ok with us just removing the API